core/libdivecomputer.c

```diff
@@ -151,7 +151,10 @@ static int parse_gasmixes(device_data_t *devdata, struct dive *dive, dc_parser_t
 			cyl.gasmix.he.permille = he;
 		}
 
-		if (dive->dc.divemode == CCR)
+		if (rc != DC_STATUS_SUCCESS)
```

Comment:
Wouldn't it be more logical to test for `rc == DC_STATUS_UNSUPPORTED`?

Reply:
I was hoping to find a status that is more specific to a gasmix being inactive than `DC_STATUS_UNSUPPORTED` but was not successful.
I am not aware of any other case where libdivecomputer is returning `DC_STATUS_UNSUPPORTED` for a gasmix, but this is the only return value except for `DC_STATUS_SUCCESS` where we are not [returning an error](https://github.com/subsurface/subsurface/pull/3938/files#diff-ce0c7332cc0e244f78ce7d404d86f881f69e7f537c043e64f0ad4b1547d1d105L129), so doing it this way seemed to capture other cases as well.

Reply:
> I was hoping to find a status that is more specific to a gasmix being inactive than DC_STATUS_UNSUPPORTED but was not successful.

I guess you should discuss this with @jefdriesen?